File: webapp/src/app/model/fixture.ts
import {SimpleLeague} from './league';

export interface Fixture {
  fixtureId: number;
  leagueId: number;
  startsAt: number;
  startsAtDay: string;
  startsAtTime: string;
  venue: string;
  homeTeam: SimpleTeam;
  awayTeam: SimpleTeam;
}

export interface FixturesAtDate {
  date: string;
  fixtures: Fixture[];
}

export interface UserLeaguesFixtures {
  fixturesByDate: FixturesAtDate[];
  leagues: SimpleLeague[];
}

export interface SimpleTeam {
  id: number;
  name: string;
  logoUrl: string;
}

File: main/src/main/java/com/smartbet/demo/fixtures/domain/Fixture.java
package com.smartbet.demo.fixtures.domain;

import com.smartbet.demo.team.domain.SimpleTeam;
import lombok.Data;

import java.time.Instant;

@Data
public class Fixture {
    private final int fixtureId;
    private final int leagueId;
    private final Instant startsAt;
    private final String startsAtDay;
    private final String startsAtTime;
    private final String venue;
    private final SimpleTeam homeTeam;
    private final SimpleTeam awayTeam;
}
